src/util.rs

```diff
@@ -64,14 +67,40 @@ where
     }
 }
 
+pub fn validate_dim(raw_dim: &[u16; 8]) -> Result<&[u16]> {
+    let ndim = validate_dimensionality(raw_dim)?;
+    let o = &raw_dim[1..ndim + 1];
+    if let Some(i) = o.iter().position(|&x| x == 0) {
+        return Err(NiftiError::InconsistentDim(i as u8, raw_dim[i]));
+    }
+    Ok(o)
+}
+
+pub fn validate_dimensionality(raw_dim: &[u16; 8]) -> Result<usize> {
+    if raw_dim[0] == 0 || raw_dim[0] > 7 {
+        return Err(NiftiError::InconsistentDim(0, raw_dim[0]));
+    }
+    Ok(usize::from(raw_dim[0]))
```

Comment:
Why not use the good old `as usize`? Isn't it identical?

Reply:
Not exactly. `usize::from` does not allow a conversion from a wider type, whereas a cast using `as` can truncate to narrower types. Using `as` in cases where the conversion should be lossless [triggers a Clippy lint](https://rust-lang.github.io/rust-clippy/current/index.html#cast_lossless).